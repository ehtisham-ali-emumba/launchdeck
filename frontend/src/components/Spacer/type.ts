export type SpacerProps = {
  marginTop?: string;
  marginBottom?: string;
  marginLeft?: string;
  marginRight?: string;
  paddingTop?: string;
  paddingBottom?: string;
  paddingLeft?: string;
  paddingRight?: string;
};
