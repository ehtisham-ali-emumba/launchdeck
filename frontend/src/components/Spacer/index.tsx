import styled from "styled-components";
import type { SpacerProps } from "./type";

export const Spacer = styled.div<SpacerProps>`
  margin-top: ${(props) => props.marginTop || "0"};
  margin-bottom: ${(props) => props.marginBottom || "0"};
  margin-left: ${(props) => props.marginLeft || "0"};
  margin-right: ${(props) => props.marginRight || "0"};
  padding-top: ${(props) => props.paddingTop || "0"};
  padding-bottom: ${(props) => props.paddingBottom || "0"};
  padding-left: ${(props) => props.paddingLeft || "0"};
  padding-right: ${(props) => props.paddingRight || "0"};
  height: ${(props) => props.top || "0"};
`;

export const InlineSpacer = styled.span<SpacerProps>`
  margin-top: ${(props) => props.marginTop || "0"};
  margin-bottom: ${(props) => props.marginBottom || "0"};
  margin-left: ${(props) => props.marginLeft || "0"};
  margin-right: ${(props) => props.marginRight || "0"};
  padding-top: ${(props) => props.paddingTop || "0"};
  padding-bottom: ${(props) => props.paddingBottom || "0"};
  padding-left: ${(props) => props.paddingLeft || "0"};
  padding-right: ${(props) => props.paddingRight || "0"};
  height: ${(props) => props.top || "0"};
`;
