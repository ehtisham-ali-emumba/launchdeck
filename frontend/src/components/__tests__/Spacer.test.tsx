import { render } from "@testing-library/react";

import { Spacer, InlineSpacer } from "../Spacer";

describe("Spacer", () => {
  it("applies margin and padding props correctly", () => {
    const { container } = render(
      <Spacer
        marginTop="10px"
        marginBottom="5px"
        paddingLeft="8px"
        paddingRight="12px"
      />
    );
    const spacer = container.firstChild;
    expect(spacer).toHaveStyle({
      marginTop: "10px",
      marginBottom: "5px",
      paddingLeft: "8px",
      paddingRight: "12px",
    });
  });

  it("applies top prop as height", () => {
    const { container } = render(<Spacer top="20px" />);
    const spacer = container.firstChild;
    expect(spacer).toHaveStyle({ height: "20px" });
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Spacer marginTop="1rem" paddingBottom="2rem" />
    );
    expect(container).toMatchSnapshot();
  });
});

describe("InlineSpacer", () => {
  it("applies margin and padding props correctly", () => {
    const { container } = render(
      <InlineSpacer
        marginLeft="4px"
        marginRight="6px"
        paddingTop="3px"
        paddingBottom="7px"
      />
    );
    const spacer = container.firstChild;
    expect(spacer).toHaveStyle({
      marginLeft: "4px",
      marginRight: "6px",
      paddingTop: "3px",
      paddingBottom: "7px",
    });
  });

  it("applies top prop as height", () => {
    const { container } = render(<InlineSpacer top="15px" />);
    const spacer = container.firstChild;
    expect(spacer).toHaveStyle({ height: "15px" });
  });

  it("matches snapshot", () => {
    const { container } = render(
      <InlineSpacer marginLeft="2rem" paddingRight="1rem" />
    );
    expect(container).toMatchSnapshot();
  });
});
