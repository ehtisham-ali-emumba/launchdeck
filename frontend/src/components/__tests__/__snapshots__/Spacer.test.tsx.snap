// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InlineSpacer > matches snapshot 1`] = `
<div>
  <div
    class="sc-fYsHOw eidifT"
    data-testid="spacer"
    marginleft="2rem"
    paddingright="1rem"
  />
</div>
`;

exports[`Spacer > matches snapshot 1`] = `
<div>
  <div
    class="sc-Qotzb fTRIdJ"
    data-testid="spacer"
    margintop="1rem"
    paddingbottom="2rem"
  />
</div>
`;
